pkg/network/dhcpconfigurators/common.go

```diff
@@ -0,0 +1,58 @@
+package dhcpconfigurators
+
+import (
+	"fmt"
+	"os"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+const dhcpStartedDirectory = "/var/run/kubevirt-private"
+
+type Configurator interface {
+	GetConfiguration(vmiSpecIface *v1.Interface, vmiSpecNetwork *v1.Network, podInterfaceName string) (*cache.DHCPConfig, error)
+	ExportNonRecoverableConfiguration(config *cache.DHCPConfig) error
+	EnsureDHCPServerStarted(podInterfaceName string, dhcpConfig cache.DHCPConfig, dhcpOptions *v1.DHCPOptions) error
+	getDHCPStartedFilePath(podInterfaceName string) string
```

Comment:
I've worked really hard to set this straight, and push this behavior to the correct struct. I see now you're attempting to bring this back in. I don't understand why, but, imo, the encapsulation of this interface is wrong, as can be seen by the copied implementations of each method.